### ISSUE TITLE
Deprecation warning

### ISSUE THREAD

**Issue by supersnager** [egoist/rollup-plugin-postcss#258]
My config:
```javascrtipt
postcss( {
        modules: false,  
        extract: true,  
        extensions: [ ".css" ],  
        minimize: true,  
        sourceMap: true  
      })
````

This emits deprecation warning from rollup:

(!) A plugin is directly adding properties to the bundle object in the "generateBundle" hook. This is deprecated and will be removed in a future Rollup version, please use "this.emitFile" instead.

**Comment by frederikhors** [comment]
Same here.

**Comment by Himself65** [comment]
Relate: https://github.com/egoist/rollup-plugin-postcss/commit/1170e89e5508222730c4fba1f8f2ed2acae28708 https://github.com/egoist/rollup-plugin-postcss/pull/244

**Comment by MrJohz** [comment]
> Relate: [1170e89](https://github.com/egoist/rollup-plugin-postcss/commit/1170e89e5508222730c4fba1f8f2ed2acae28708) #244

It looks like you reverted the change that fixed this - can you explain what was wrong with that PR?

**Comment by Himself65** [comment]
> It looks like you reverted the change that fixed this - can you explain what was wrong with that PR?

https://github.com/egoist/rollup-plugin-postcss/issues/250

**Comment by MrJohz** [comment]
Aha, thanks!  To fix that, would some sort of migration path make sense?  E.g:

`extract: true` -> use `this.emitFile` and avoid the deprecation warning
`extract:'string'` -> use the existing `bundle[filename]` assignment.

Alternatively, a new configuration key could be given: `emit` (which is just a boolean value).  It would be an error to configure both `emit` and `extract` at the same time.

I think adding the new configuration key is the only way to avoid any breaking changes, but it also adds more configuration options, which is also difficult.  I'm happy to put together a pull request for any of these options - which do you think makes the most sense?

**Comment by Himself65** [comment]
I would like to use `this.emitFile`, but there have some incompatible as you can see. but I think we just need to resolve the `path` to compatible with the `emitFile`, it will work. BTW, I'm very welcome you to open a PR.

**Comment by gavinsharp** [comment]
Just linking this PR since it is related (outdated; possibly incomplete given issues mentioned here earlier): https://github.com/egoist/rollup-plugin-postcss/pull/197

**Comment by inno76tink** [comment]
hey deprecate message patch  

node_modules/rollup-plugin-postcss/dist/index.js

1.  line 821 change 
from code :
 generateBundle(options_, bundle) {

to code
    generateBundle(options_, bundle) {
      var _this = this;

2. line 905 change
from code :
        const codeFile = {
          fileName: codeFileName,
          isAsset: true,
          source: code
        };
        bundle[codeFile.fileName] = codeFile;

        if (map) {
          const mapFile = {
            fileName: mapFileName,
            isAsset: true,
            source: map
          };
          bundle[mapFile.fileName] = mapFile;

to code :
        const codeFile = {
          fileName: codeFileName,
          type: 'asset',
          source: code
        };
 _this.emitFile(codeFile)
        if (map) {
          const mapFile = {
            fileName: mapFileName,
            type: 'asset',
            source: map
          };
 _this.emitFile(mapFile)

==================================================
If you do this, the deprecate message is no longer printed.

**Comment by vladshcherbin** [comment]
Please use https://github.com/egoist/rollup-plugin-postcss/pull/197 solution or other fix, this warning in watch mode looks extremely bad, it is printed on every rebuild:

![image](https://user-images.githubusercontent.com/6711845/79611898-4246a600-8104-11ea-9875-de63a95d48c5.png)


🙏

**Comment by SASUKE40** [comment]
> Please use #197 solution or other fix, this warning in watch mode looks extremely bad, it is printed on every rebuild:
> 
> ![image](https://user-images.githubusercontent.com/6711845/79611898-4246a600-8104-11ea-9875-de63a95d48c5.png)
> 
> 🙏

The rollup v2 adaptation  is still work in progress and will be released soon to resolve this warning.

**Comment by SASUKE40** [comment]
It is solved in `v2.8.1` via upgrading rollup to v2.

**Comment by vladshcherbin** [comment]
@SASUKE40 It's gone, thank you ❤️
